tools/const_finder/test/const_finder_test.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' show jsonEncode;
+import 'dart:io';
+
+import 'package:const_finder/const_finder.dart';
+import 'package:path/path.dart' as path;
+
+int exitCode = 0;
+
+void expect<T>(T value, T expected) {
+  if (value != expected) {
+    stderr.writeln('Expected: $expected');
+    stderr.writeln('Actual:   $value');
+    exitCode = -1;
+  }
+}
+
+final String basePath =
+    path.canonicalize(path.join(path.dirname(Platform.script.path), '..'));
+final String fixtures = path.join(basePath, 'test', 'fixtures');
+final String consts = path.join(fixtures, 'lib', 'consts.dart');
+final String constsAndNon = path.join(fixtures, 'lib', 'consts_and_non.dart');
+final String constsDill = path.join(fixtures, 'consts.dill');
```

Comment:
I would consider doing this in a temp directory, only if the bots are still having caching issues

Reply:
I'm worried about the test failing to delete the temp dir.  I just added this extension to .gitignore.